src/osyris/spatial/spatial.py

```diff
@@ -0,0 +1,43 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/nvaytet/osyris)
+
+import numpy as np
+
+
+def extract_sphere(dataset, radius, origin):
+    """
+    Extract a spherical subdomain around an origin point.
+    """
+    subdomain = dataset.__class__(nout=dataset.meta["nout"], path=dataset.meta["path"])
+    subdomain.meta = dataset.meta
```

Comment:
I think we might need to explicitly `copy` here, to make sure our subdomain is a standalone Dataset in its own right.